upload/lambdas/health_check/health_check.py

```diff
@@ -0,0 +1,109 @@
+        deadletter_results = self.get_deadletter_count()
+        undeleted_upload_area_count = self.parse_and_query_db(self.undeleted_areas_count_query)
+        checksumming_areas = self.parse_and_query_db(self.checksumming_areas_count_query)
+        validating_areas = self.parse_and_query_db(self.validation_areas_count_query)
+        status_info = f"DEADLETTER_QUEUE: {int(deadletter_results['ApproximateNumberOfMessagesVisible'])} in queue, " \
+                      f"{int(deadletter_results['NumberOfMessagesReceived'])} added in past 24 hrs\n" \
+                      f"UPLOAD_AREAS: {undeleted_upload_area_count} undeleted areas, {checksumming_areas} stuck in " \
+                      f"checksumming, {validating_areas} stuck in validation"
+
+        attachments = [{
+            "title": f"Health Check Report for {self.env}:",
+            "color": "good",
+            "text": status_info
+        }]
+
+        self.post_message_to_url(self.webhook, {"attachments": attachments})
+
+    def post_message_to_url(self, url, message):
+        body = json.dumps(message)
+        headers = {'Content-Type': 'application/json'}
+        requests.post(url=url, data=body, headers=headers)
+
+    def get_deadletter_count(self):
+        now = datetime.utcnow()
+        yesterday = now - timedelta(hours=24)
+        metric_data_queries = [
+            {
+                'Id': 'visible_messages',
+                'MetricStat': {
+                    'Metric': {
+                        'Namespace': 'AWS/SQS',
+                        'MetricName': 'ApproximateNumberOfMessagesVisible',
+                        'Dimensions': [
+                            {
+                                'Name': 'QueueName',
+                                'Value': f'dcp-upload-pre-csum-deadletter-queue-{self.env}'
+                            }
+                        ]
+                    },
+                    'Period': 90000,
+                    'Stat': 'Average'
+                }
+            },
+            {
+                'Id': 'recieved_messages',
+                'MetricStat': {
+                    'Metric': {
+                        'Namespace': 'AWS/SQS',
+                        'MetricName': 'NumberOfMessagesReceived',
+                        'Dimensions': [
+                            {
+                                'Name': 'QueueName',
+                                'Value': f'dcp-upload-pre-csum-deadletter-queue-{self.env}'
+                            }
+                        ]
+                    },
+                    'Period': 90000,
+                    'Stat': 'Average'
+                }
+            }
+        ]
+        response = client.get_metric_data(MetricDataQueries=metric_data_queries, StartTime=yesterday, EndTime=now)
+        results = {}
+        for info in response['MetricDataResults']:
+            results[info['Label']] = info['Values'][0]
+        return results
+
+    def parse_and_query_db(self, query):
```

Comment:
rename per comment above. (nit) maybe query_db_and_return_first_row since only count queries are running through this function?

Reply:
👍